### ISSUE TITLE
Arrays within Arrays on multiple HTML elements, only the last one is used

### ISSUE THREAD

**Issue by boardy** [webdevelopers-eu/jquery-lazy-template#2]
Hi, 

I have an array data coming back from my API, within my data is another Array which contains members which I want to display in separate `td` within a `table`. 

The data looks like the following:

```
"data": [
	{
		"CallType": "Inbound",
		"CallFrom": "01234567",
		"CallState": "Answered",
		"Date": "21\/07\/2017",
		"Time": "09:10:40",
		"CustomerDetails": {
			"Customer": "Customer Name",
			"Name": "Contact Name",
			"Email": "someone@example.com"
		}
	}
]
```

I am using the template as follows

`$('[template="call_log_template"]').template(object.data);`

My HTML looks like:

```
<table class="table table-striped">
                <tr class="titleRow">
                    <th>Date Called</th>
                    <th>Time Called</th>
                    <th>Call From</th>
                    <th>Customer</th>
                    <th>Contact</th>
                    <th>Email</th>
                    <th>Call State</th>
                </tr>
                <tr template="call_log_template">
                    <td z-var="Date ."></td>
                    <td z-var="Time ."></td>
                    <td z-var="CallFrom ."></td>
                    <td template="[CustomerDetails]" z-var="Customer ."></td>
                    <td template="[CustomerDetails]" z-var="Name ."></td>
                    <td template="[CustomerDetails]" z-var="Email ."></td>
                    <td z-var="CallState ."></td>
                </tr>
            </table>
```

The issue is that when the table is shown in the 4th column where it should show the customers name, it seems to get skipped over, as does the name, and then only shows the email, followed by the call state column. The customer and the name aren't displayed, so I'm only outputting 5 columns instead of 7. 

I've gone through the tutorial but I can't see a way around this at the moment.

Thanks

Chris

**Comment by webdevelopers-eu** [comment]
Hello Chris,

here is the [CodePen Example](https://codepen.io/anon/pen/BdowNZ?editors=1010).

Note that `template="[CustomerDetails]"` has the meaning of _for each item inside CustomerDetails_ - so in your case it should be placed rather on `tr` element then on `td`.

**Comment by webdevelopers-eu** [comment]
I am closing this issue as this is not a bug. If I missed something reopen it.

**Comment by boardy** [comment]
Hi, 

I'm not sure if I'm missing something, but from your example, this now means that the only things that do get set are the variables within the CustomerDetails array so its now only filled in 3 columns and the others are blank. 

What I need to do is columns 1, 2 and 3 are set using the members of the root data array, then 4, 5 and 6 are then set using the members of the CustomerDetails array, then column 7 is set using the member again from the root of the data array. 

Unless I'm missing something it doesn't seem like what I need is supported, as arrays within the array can be only used as part of another sub template, which I can then provide children elements to set the variables, which in my case I can't do as the template is set for the data array, I then can't get the values out of CustomerDetails to then set each TD when needed. 

Thanks

Chris

**Comment by webdevelopers-eu** [comment]
Ah, I see!

No, that is not supported because the pre-processing of data are in many cases trivial in Javascript so I didn't consider support for deep referencing of variables (I may if it will be the common use case).

For now it is indeed not supported but it takes just one-liner to achieve that effect:

```javascript
response.data = response.data.map((v) => $.extend(v, v.CustomerDetails));
```
I have updated the [Codepen](https://codepen.io/anon/pen/BdowNZ?editors=1010) example. I also assumed that response.data array will contain more items so I have fixed it so it works with multiple lines. 

Check it out and let me know if that's it.
